tower-reconnect/src/lib.rs

```diff
@@ -4,16 +4,17 @@ extern crate log;
 extern crate tower_service;
 
 use futures::{Future, Async, Poll};
-use tower_service::{Service, NewService};
+use tower_service::{Service, MakeService};
 
 use std::{error, fmt};
 
-pub struct Reconnect<T, Request>
+pub struct Reconnect<T, Ctx, Request>
 where
-    T: NewService<Request>,
+    T: MakeService<Ctx, Request>,
 {
-    new_service: T,
-    state: State<T, Request>,
+    context: Ctx,
```

Comment:
If we are bike shedding, I prefer "target" over "context". I feel that it makes the intent a bit more clear.